<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="srm">
    <group name="srm_base_l">
        <joint name="left_base_joint"/>
        <joint name="left_summit_base_footprint_joint"/>
        <joint name="left_summit_back_left_wheel_joint"/>
        <joint name="left_summit_back_right_wheel_joint"/>
        <joint name="left_summit_front_left_wheel_joint"/>
        <joint name="left_summit_front_right_wheel_joint"/>
    </group>
    <group name="srm_l_ur">
        <joint name="ur_left_base_link-base_link_inertia"/>
        <joint name="ur_left_shoulder_pan_joint"/>
        <joint name="ur_left_shoulder_lift_joint"/>
        <joint name="ur_left_elbow_joint"/>
        <joint name="ur_left_wrist_1_joint"/>
        <joint name="ur_left_wrist_2_joint"/>
        <joint name="ur_left_wrist_3_joint"/>
        <chain base_link="ur_left_base_link_inertia" tip_link="ur_left_wrist_3_link"/>
    </group>
    <group name="srm_l_ee">
        <link name="ur_left_robotiq_hand"/>
        <link name="ur_left_finger_left"/>
        <link name="ur_left_finger_right"/>
        <link name="ur_left_finger_tip"/>
        <chain base_link="ur_left_robotiq_hand" tip_link="ur_left_finger_right"/>
    </group>
    
    <group name="srm_l">
        <group name="srm_base_l"/>
        <group name="srm_l_ur"/>
        <group name="srm_l_ee"/>
    </group>
    <group_state name="home_l" group="srm_l_ur">
<!-- 
        <joint name="left_summit_back_left_wheel_joint" value="0"/>
        <joint name="left_summit_back_right_wheel_joint" value="0"/>
        <joint name="left_summit_front_left_wheel_joint" value="0"/>
        <joint name="left_summit_front_right_wheel_joint" value="0"/>
         -->
        <joint name="ur_left_elbow_joint" value="-2.7601"/>
        <joint name="ur_left_shoulder_lift_joint" value="0.305"/>
        <joint name="ur_left_shoulder_pan_joint" value="-1.57"/>
        <joint name="ur_left_wrist_1_joint" value="-1.3893"/>
        <joint name="ur_left_wrist_2_joint" value="1.57"/>
        <joint name="ur_left_wrist_3_joint" value="0"/>
    </group_state>
    <group_state name="home_l_base_ur_ee" group="srm_l_base_ur_ee">
        <joint name="left_summit_back_left_wheel_joint" value="0"/>
        <joint name="left_summit_back_right_wheel_joint" value="0"/>
        <joint name="left_summit_front_left_wheel_joint" value="0"/>
        <joint name="left_summit_front_right_wheel_joint" value="0"/>
        <joint name="ur_left_elbow_joint" value="-2.7601"/>
        <joint name="ur_left_shoulder_lift_joint" value="0.305"/>
        <joint name="ur_left_shoulder_pan_joint" value="-1.57"/>
        <joint name="ur_left_wrist_1_joint" value="-1.3893"/>
        <joint name="ur_left_wrist_2_joint" value="1.57"/>
        <joint name="ur_left_wrist_3_joint" value="0"/>
        <!-- <joint name="ur_right_base_link-base_link_inertia" value="0"/> -->
    </group_state>

    <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
    <end_effector name="ee_l" parent_link="ur_left_wrist_3_link" group="srm_l_ee" parent_group="srm_l_ur"/>
    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_back_right_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_base_link" reason="Adjacent"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_front_laser_base_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="left_summit_rear_laser_base_link" reason="Never"/>
    
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="ur_left_shoulder_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_left_wheel_link" link2="ur_left_upper_arm_link" reason="Never"/>


    <disable_collisions link1="left_summit_back_right_wheel_link" link2="left_summit_base_link" reason="Adjacent"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="left_summit_front_laser_base_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="left_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="left_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="left_summit_rear_laser_base_link" reason="Never"/>

    <disable_collisions link1="left_summit_back_right_wheel_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="ur_left_shoulder_link" reason="Never"/>
    <disable_collisions link1="left_summit_back_right_wheel_link" link2="ur_left_upper_arm_link" reason="Never"/>
    
    <disable_collisions link1="left_summit_base_link" link2="left_summit_front_laser_base_link" reason="Adjacent"/>
    <disable_collisions link1="left_summit_base_link" link2="left_summit_front_left_wheel_link" reason="Adjacent"/>
    <disable_collisions link1="left_summit_base_link" link2="left_summit_front_right_wheel_link" reason="Adjacent"/>
    <disable_collisions link1="left_summit_base_link" link2="left_summit_rear_laser_base_link" reason="Adjacent"/>

    <disable_collisions link1="left_summit_base_link" link2="ur_left_base_link_inertia" reason="Adjacent"/>
    <disable_collisions link1="left_summit_base_link" link2="ur_left_shoulder_link" reason="Never"/>
    
    <disable_collisions link1="left_summit_front_laser_base_link" link2="left_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_front_laser_base_link" link2="left_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_front_laser_base_link" link2="left_summit_rear_laser_base_link" reason="Never"/>

    <disable_collisions link1="left_summit_front_laser_base_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_front_laser_base_link" link2="ur_left_shoulder_link" reason="Never"/>

    <disable_collisions link1="left_summit_front_left_wheel_link" link2="left_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="left_summit_front_left_wheel_link" link2="left_summit_rear_laser_base_link" reason="Never"/>

    <disable_collisions link1="left_summit_front_left_wheel_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_front_left_wheel_link" link2="ur_left_shoulder_link" reason="Never"/>
    <disable_collisions link1="left_summit_front_left_wheel_link" link2="ur_left_upper_arm_link" reason="Never"/>
    
    <disable_collisions link1="left_summit_front_right_wheel_link" link2="left_summit_rear_laser_base_link" reason="Never"/>

    <disable_collisions link1="left_summit_front_right_wheel_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_front_right_wheel_link" link2="ur_left_shoulder_link" reason="Never"/>
    <disable_collisions link1="left_summit_front_right_wheel_link" link2="ur_left_upper_arm_link" reason="Never"/>

    <disable_collisions link1="left_summit_rear_laser_base_link" link2="ur_left_base_link_inertia" reason="Never"/>
    <disable_collisions link1="left_summit_rear_laser_base_link" link2="ur_left_shoulder_link" reason="Never"/>

    <disable_collisions link1="ur_left_base_link_inertia" link2="ur_left_shoulder_link" reason="Adjacent"/>
    
    <disable_collisions link1="ur_left_finger_left" link2="ur_left_finger_right" reason="Adjacent"/>
    <disable_collisions link1="ur_left_finger_left" link2="ur_left_robotiq_hand" reason="Adjacent"/>
    <disable_collisions link1="ur_left_finger_left" link2="ur_left_wrist_1_link" reason="Never"/>
    <disable_collisions link1="ur_left_finger_left" link2="ur_left_wrist_2_link" reason="Never"/>
    <disable_collisions link1="ur_left_finger_left" link2="ur_left_wrist_3_link" reason="Never"/>

    <disable_collisions link1="ur_left_finger_right" link2="ur_left_robotiq_hand" reason="Default"/>
    <disable_collisions link1="ur_left_finger_right" link2="ur_left_wrist_1_link" reason="Never"/>
    <disable_collisions link1="ur_left_finger_right" link2="ur_left_wrist_2_link" reason="Never"/>
    <disable_collisions link1="ur_left_finger_right" link2="ur_left_wrist_3_link" reason="Never"/>
    <disable_collisions link1="ur_left_forearm_link" link2="ur_left_upper_arm_link" reason="Adjacent"/>
    <disable_collisions link1="ur_left_forearm_link" link2="ur_left_wrist_1_link" reason="Adjacent"/>
    <disable_collisions link1="ur_left_forearm_link" link2="ur_left_wrist_2_link" reason="Never"/>

    <disable_collisions link1="ur_left_robotiq_hand" link2="ur_left_wrist_1_link" reason="Never"/>
    <disable_collisions link1="ur_left_robotiq_hand" link2="ur_left_wrist_2_link" reason="Never"/>
    <disable_collisions link1="ur_left_robotiq_hand" link2="ur_left_wrist_3_link" reason="Adjacent"/>
    <disable_collisions link1="ur_left_shoulder_link" link2="ur_left_upper_arm_link" reason="Adjacent"/>
    <disable_collisions link1="ur_left_wrist_1_link" link2="ur_left_wrist_2_link" reason="Adjacent"/>
    <disable_collisions link1="ur_left_wrist_1_link" link2="ur_left_wrist_3_link" reason="Never"/>
    <disable_collisions link1="ur_left_wrist_2_link" link2="ur_left_wrist_3_link" reason="Adjacent"/>

</robot>